internal/graphapi/validation.go

```diff
@@ -0,0 +1,38 @@
+package graphapi
+
+import (
+	"regexp"
+	"strings"
+
+	"go.infratographer.com/x/gidx"
+)
+
+// validateGidx validates a gidx.PrefixedID
+func validateGidx(gid gidx.PrefixedID) error {
+	if _, err := gidx.Parse(gid.String()); err != nil {
+		return err
+	}
+
+	id := strings.TrimSpace(gid.String())
+	if len(id) > 0 {
+		if !regexp.MustCompile(`^[a-zA-Z0-9_-]+$`).MatchString(id) {
+			return ErrInvalidCharacters
+		}
+	} else {
+		return ErrFieldEmpty
+	}
+
+	return nil
+}
+
+// sanitizeField sanitizes a field string
+func sanitizeField(field string) string {
+	s := strings.TrimSpace(field)
+	// s = template.HTMLEscapeString(s)
```

Comment:
did you want to enable this commented out code?

Reply:
I do a similar validation here: https://github.com/infratographer/load-balancer-api/pull/357/files#diff-f15d646ad33e9ea909784b97e72caf5674dbd10d775d281cc663df80d5947cf3R70-R79

so left that commented just in case, but can remove it if its cleaner